flutter/news/test/core/network/network_info_test.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:internet_connection_checker/internet_connection_checker.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:news/core/network/network_info.dart';
+
+import 'network_info_test.mocks.dart';
+
+@GenerateMocks([InternetConnectionChecker])
+void main() {
+  late NetworkInfoImpl networkInfoImpl;
+  late MockInternetConnectionChecker mockInternetConnectionChecker;
+
+  setUp(() {
+    mockInternetConnectionChecker = MockInternetConnectionChecker();
+    networkInfoImpl = NetworkInfoImpl(mockInternetConnectionChecker);
+  });
+
+  group("isConnected", () {
+    test(
+      'should forward the call to InternetConnectChecker.hasConnection',
```

Comment:
I think you need to tests 2 cases. I would name them as:


`GIVEN device has connection WHEN checking if device has connection THEN returns true`

`GIVEN device does not have connection WHEN checking if device has connection THEN returns false`

Your test is for the positive part.


Without this two tests, you can go to the `network_info` class, make it `return` true on every case and the test will still pass

Reply:
i am actually not testing for connection, i am testing if the call is actually forwarded to another lib, by returning the future and not the value itself.

however will also had the tests you mentioned because you are correct

Reply:
updated on https://github.com/novoda/multiplatform-playground/commit/8323a52c3c021a85d1662fa2609c061534a5fce3